pkgs/desktops/cinnamon/nemo/0001-patch-datadir.patch

```diff
@@ -0,0 +1,34 @@
+From f0fe6625fb607f9ecaa86cb2c6ba233b13bb2f25 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Maciej=20Kr=C3=BCger?= <mkg20001@gmail.com>
+Date: Thu, 28 Nov 2019 22:02:09 +0100
+Subject: [PATCH 1/2] patch: datadir
+
+---
+ libnemo-extension/meson.build | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/libnemo-extension/meson.build b/libnemo-extension/meson.build
+index a177ed6..a3708c9 100644
+--- a/libnemo-extension/meson.build
++++ b/libnemo-extension/meson.build
+@@ -64,7 +64,7 @@ gnome.generate_gir(nemo_extension_lib,
+   namespace: 'Nemo',
+   includes: [ 'Gtk-3.0', 'Gio-2.0', 'GLib-2.0', ],
+   include_directories: [ rootInclude, ],
+-  install_dir_typelib: join_paths(go_intr.get_pkgconfig_variable('libdir'), 'girepository-1.0'),
++  install_dir_typelib: join_paths(get_option('libdir'), 'girepository-1.0'),
+   install: true,
+ )
+ 
+@@ -76,7 +76,7 @@ if meson.version().version_compare('>=0.41.0')
+     requires: [ 'gio-2.0', 'glib-2.0', 'gtk+-3.0' ],
+     libraries: nemo_extension_lib,
+     subdirs: 'nemo',
+-    variables: 'extensiondir=${libdir}/@0@/@1@'.format('nemo', 'extensions-3.0'),
++    variables: 'extensiondir=/run/current-system/sw/lib/@0@/@1@'.format('nemo', 'extensions-3.0'),
```

Comment:
About this: Shouldn't it go into current-system/share?
Or should we do a completly different approach and use a symlinkJoin, add a plugins[<package>] argument and put that path in here? (That would recompile nemo whenever the plugins change but wouldn't be impure)

Or something else like wrapper that passes a path via ENV?

Reply:
So installed extensions I guess will get linked at `/run/current-system/sw/lib/nemo/extensions-3.0` and NixOS will link things at `lib`.

It's very popularized within nixos to patch file managers with extensions like
* https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/xfce/core/thunar/thunarx_plugins_directory.patch

so we can set which directory to load extensions from via an environment variable.
And as you've mentioned, we'd use the `symlinkJoin` wrapper 
* https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/xfce/core/thunar/wrapper.nix

and some glue in the nixos module
https://github.com/NixOS/nixpkgs/blob/a65f338ccbd679c1d9f6886d17758d99c80cd938/nixos/modules/services/x11/desktop-managers/xfce.nix#L45

Reply:
I happen to really like the above ^, but patching the extension loading core isn't for everybody and could be risky. How many extensions are there in the nemo ecosystem?

Reply:
There's this repo containing most of them: https://github.com/linuxmint/nemo-extensions

As for extensions in the wild: No idea